codalab/worker/local_run/local_dependency_manager.py

```diff
@@ -109,19 +112,32 @@ def _load_state(self):
                         dep_key, dep_state.path, state['paths']
                     )
                 )
-        for path in state['paths']:
-            full_path = os.path.join(self.dependencies_dir, path)
-            if os.path.exists(full_path):
-                paths.add(path)
-            else:
-                logger.info(
-                    "Path {} in loaded state but doesn't exist in the filesystem".format(full_path)
+
+        state_paths = state['paths']
+        # Find out the valid loaded paths by intersecting them with local_directories.
+        paths = state_paths.intersection(local_directories)
+        # Filter in those invalid loaded paths and generate logging message
+        invalid_paths = state_paths - paths
+        for path in invalid_paths:
+            logger.info(
+                "Path {} in loaded state but doesn't exist in the filesystem".format(
+                    os.path.join(self.dependencies_dir, path)
                 )
+            )
+        # Remove the redundant dependencies from local file system
```

Comment:
These aren't 'redundant' right?  These are 'orphaned' (not accounted for in dependencies)?